### PR TITLE
Build steemd with boost 1.69

### LINKED ISSUE #3253: make not working with boost 1.68
Environment: Ubuntu 16.04
When I'm running `make`, I get the following output:
 [19%] Building CXX object libraries/fc/tests/CMakeFiles/all_tests.dir/all_tests.cpp.o
In file included from /root/opt/boost_1_67_0/include/boost/test/unit_test_log.hpp:25:0,
                 from /root/opt/boost_1_67_0/include/boost/test/tools/old/impl.hpp:19,
                 from /root/opt/boost_1_67_0/include/boost/test/test_tools.hpp:46,
                 from /root/opt/boost_1_67_0/include/boost/test/unit_test.hpp:18,
                 from /root/steem/steem/libraries/fc/tests/all_tests.cpp:2:
/root/opt/boost_1_67_0/include/boost/test/unit_test_log.hpp:227:28: error: ‘boost::unit_test::{anonymous}::unit_test_log’ defined but not used [-Werror=unused-variable]
 BOOST_TEST_SINGLETON_INST( unit_test_log )
                            ^
/root/opt/boost_1_67_0/include/boost/test/utils/trivial_singleton.hpp:67:36: note: in definition of macro ‘BOOST_TEST_SINGLETON_INST’
 namespace { BOOST_JOIN( inst, _t)& inst = BOOST_JOIN( inst, _t)::instance(); }
                                    ^
cc1plus: all warnings being treated as errors
libraries/fc/tests/CMakeFiles/all_tests.dir/build.make:62: recipe for target 'libraries/fc/tests/CMakeFiles/all_tests.dir/all_tests.cpp.o' failed
make[2]: *** [libraries/fc/tests/CMakeFiles/all_tests.dir/all_tests.cpp.o] Error 1
CMakeFiles/Makefile2:420: recipe for target 'libraries/fc/tests/CMakeFiles/all_tests.dir/all' failed
make[1]: *** [libraries/fc/tests/CMakeFiles/all_tests.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

### PR DESCRIPTION
**Pull request by sgerbino**
Resolves #3253.

### CODE REVIEW

File: libraries/fc/src/thread/context.hpp [hunk 1]
```diff
@@ -14,6 +13,12 @@
 #define BOOST_COROUTINES_NO_DEPRECATION_WARNING // Boost 1.61
 #define BOOST_COROUTINE_NO_DEPRECATION_WARNING // Boost 1.62
 
+#if BOOST_VERSION >= 106800
+#include <boost/context/continuation_fcontext.hpp>
+#else
+#include <boost/context/all.hpp>
```
Comment (mvandeberg) [inline libraries/fc/src/thread/context.hpp]: This breaks building on boost < 1.68.

The diagnostic pragmas from above need to be moved as well.

**Comment by mvandeberg** [comment]
Building on OS X succeeds, but outputs linker warnings of the form:

`ld: warning: direct access in function 'boost::system::detail::system_error_category::default_error_condition(int) const' from file '$HOME/Library/Homebrew/opt/boost/lib/libboost_filesystem-mt.a(operations.o)' to global weak symbol 'boost::system::detail::cat_holder<void>::system_category_instance' from file 'CMakeFiles/appbase_example.dir/main.cpp.o' means the weak symbol cannot be overridden at runtime. This was likely caused by different translation units being compiled with different visibility settings.`

Adding the build option `-fvisibility=hidden` resolves the warnings.

**Comment by bobinson** [comment]
@sgerbino @mvandeberg  : Did we add support for Ubuntu 1804 as well ?

**Comment by sgerbino** [comment]
> @sgerbino @mvandeberg : Did we add support for Ubuntu 1804 as well ?

Yes, there is 18.04 support

**Comment by bobinson** [comment]
Awesome @sgerbino !

**Pull request merged**